src/quic/node_quic_util.h

```diff
@@ -87,6 +87,19 @@ class StatsBase {
       v8::Local<v8::Object> wrap,
       int options = HistogramOptions::NONE);
 
+  inline ~StatsBase();
+
+  using AddField = std::function<void(const char*, uint64_t)>;
+
+  // The StatsDebug utility is used when StatsBase is destroyed
+  // to output statistical information.
+  struct StatsDebug {
+    Q* ptr;
+    explicit StatsDebug(Q* ptr_) : ptr(ptr_) {}
+    std::string ToString() const;
+    void ToString(AddField add_field) const;
```

Comment:
I realize it’s not that important for debug-only code, but typically, this would be implemented by making function argument a template (in order to avoid the overhead of `std::function` here)

Reply:
Not sure I follow

Reply:
I’m taking about something like 

```c++
    template <typename Fn>
    void ToString(Fn&& add_field);
```

which is otherwise the same but doesn’t use an `std::function` to pass the function pointer.

Like I said, this isn’t particularly important for debug-only code, so feel free to ignore.

Reply:
Got it... made some tweaks. Take a look now :-)